src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -4402,5 +4402,25 @@ public async Task EditorConfigOptions(string projectPath, string configPath, str
             var projectOptions = document.Project.GetAnalyzerConfigOptions();
             Assert.Equal(appliedToEntireProject, projectOptions?.AnalyzerOptions.TryGetValue("indent_style", out value) == true && value == "tab");
         }
+
+        [Fact]
+        public void GetRelatedDocumentsDoesNotReturnOtherTypesOfDocuments()
+        {
+            using var workspace = CreateWorkspace();
+
+            const string FilePath = "File.cs";
+
+            var solution = workspace.CurrentSolution
+                .AddProject("TestProject", "TestProject", LanguageNames.CSharp)
+                .AddDocument("File.cs", "", filePath: FilePath).Project
+                .AddAdditionalDocument("File.cs", text: "", filePath: FilePath).Project.Solution;
+
+            // GetDocumentIdsWithFilePath should return two, since it'll count all types of documents
+            Assert.Equal(2, solution.GetDocumentIdsWithFilePath(FilePath).Length);
```

Comment:
aside.  i'm slightly worried about some code possibly calling this helper directly, not realizing this case.  We might want to audit.

Reply:
example: this is broken if i'm not mistaken, right:

```
        public static ImmutableArray<Document> GetDocuments(this Solution solution, string documentPath)
        {
            var documentIds = solution.GetDocumentIdsWithFilePath(documentPath);

            // We don't call GetRequiredDocument here as the id could be referring to an additional document.
            var documents = documentIds.Select(solution.GetDocument).WhereNotNull().ToImmutableArray();
            return documents;
        }
```